sendbuffer.go

```diff
@@ -100,17 +104,46 @@ func (buf *sendBuffer) sendLoop(w io.Writer) {
 	}
 }
 
+func (buf *sendBuffer) send(b []byte, writeDeadline time.Time) (int, error) {
+	buf.muClosing.RLock()
+	n, err := buf.doSend(b, writeDeadline)
+	buf.muClosing.RUnlock()
+	return n, err
+}
+
+func (buf *sendBuffer) doSend(b []byte, writeDeadline time.Time) (int, error) {
+	if buf.closing {
+		// Make it look like the write worked even though we're not going to send it
+		// anywhere (TODO, might be better way to handle this?)
+		return len(b), nil
```

Comment:
I think the usual behavior here would be to send `EPIPE`.

Reply:
I thought it's intentionally kept for some reason. This makes sense, will change.

Reply:
Hmm it breaks `TestStreamCloseRemoteAfterEcho` which specifically covers this.
```
	assert.NoError(t, err, "We got an EOF on read, but writing should still work")
```

Reply:
Ha, you're right! That test is wrong. Writing to a connection that's been closed by the remote end should result in an EPIPE. I've updated the code to return EPIPE and fixed the test.